init/pkg/update_pod_status_util.go

```diff
@@ -46,52 +47,49 @@ func (v *Volume) getMountPoint() string {
 	return filepath.Join(fileSystemMountPoint, v.volumeName)
 }
 
-func getImageVersion(image string) (version int, err error) {
+func getImageVersion(image string) (majorVersion int, err error) {
 	ver, err := asdbv1beta1.GetImageVersion(image)
 	if err != nil {
 		return 0, err
 	}
 
 	res := strings.Split(ver, ".")
 
-	version, err = strconv.Atoi(res[0])
+	majorVersion, err = strconv.Atoi(res[0])
 	if err != nil {
 		return 0, err
 	}
 
-	return version, err
+	return majorVersion, err
 }
 
 func execute(cmd []string, stderr *os.File) error {
-	if len(cmd) > 0 {
-		var command *exec.Cmd
-		if len(cmd) > 1 {
-			command = exec.Command(cmd[0], cmd[1:]...) //nolint:gosec // internal command array
-		} else {
-			command = exec.Command(cmd[0]) //nolint:gosec // internal command array
-		}
+	if len(cmd) == 0 {
+		return nil
+	}
 
-		if stderr != nil {
-			command.Stdout = stderr
-			command.Stderr = stderr
-		} else {
-			command.Stdout = os.Stdout
-			command.Stderr = os.Stderr
-		}
+	command := exec.Command(cmd[0], cmd[1:]...) //nolint:gosec // internal command array
 
-		if err := command.Run(); err != nil {
-			return err
-		}
+	if stderr != nil {
+		command.Stdout = stderr
+		command.Stderr = stderr
+	} else {
+		command.Stdout = os.Stdout
+		command.Stderr = os.Stderr
+	}
+
+	if err := command.Run(); err != nil {
+		return err
 	}
 
 	return nil
 }
 
-func getPodImage(ctx goctx.Context, k8sClient client.Client, podNamespacedName types.NamespacedName) (string, error) {
-	logrus.Info("Get pod image ", "pod-name=", podNamespacedName)
+func (initp *InitParams) getPodImage(podNamespacedName types.NamespacedName) (string, error) {
```

Comment:
Why did we remove ctx from func param?
Ideally context should be defined once (parent context) in the top level func and the same context should be passed to every child func called instead of creating of creating context again.

See if we can incorporate this, if this is too much change then can defer this to next python-to-go PR

Reply:
made changes in this itself